libshaderc/include/shaderc.hpp

```diff
@@ -219,6 +219,20 @@ class CompileOptions {
     shaderc_compile_options_set_warnings_as_errors(options_);
   }
 
+  // Sets the weak shader kind to be used for following compilation. The
+  // shader kind set here has lowest priority compared with the
+  // forced_shader_kind passed through shaderc_compile_into_spv()
+  // and #pragma annotation in source text. The compiler will check
+  // the forced_shader_kind first. If the forced_shader_kind is
+  // shaderc_glsl_none_shader, the compiler will check the #pragma
+  // annotation in the source text. If the #pragma annotation is not set,
+  // the compiler will use the shader kind set by this function to compile
+  // the source text.
+  void SetWeakShaderKind(shaderc_shader_kind kind) {
```

Comment:
I think this would be better as a parameter to CompilerGlslToSpv, (and likewise shaderc_compile_into_spv) rather than on the compile_options object. (Meaning the shaderc_compiler_into_spv could turn into 
```
shaderc_spv_module_t shaderc_compile_into_spv(
const shaderc_compiler_t compiler,
const char* source_text,
size_t source_text_size,
shaderc_shader_kind shader_kind,

bool override_pragmas, // <<<<<

const char* entry_point_name,
const shaderc_compile_options_t additional_options);
```

That was compile_options become much more easily re-used for (as an example) compiling all shaders in a pipeline with the same options.